usb_protocol/emitters/descriptors/hid.py

```diff
@@ -0,0 +1,109 @@
+import unittest
+
+from contextlib import contextmanager
+
+from ..           import emitter_for_format
+from ..descriptor import ComplexDescriptorEmitter
+from ...types.descriptors.hid import \
+    HIDDescriptor as HIDDescriptorType
+from ...types.descriptors.hid import *
+
+ReportDescriptorEmitter = emitter_for_format(ReportDescriptor)
+
+_hid_item_length = [ 0, 1, 2, 4 ]
+
+class HIDDescriptor(ComplexDescriptorEmitter):
```

Comment:
There seem to be a bunch of naming inconsistencies, here. 

For example, this looks like it produces more than a HID descriptor (it also builds report descriptors), so perhaps this would be better called e.g. a `HIDDescriptorCollection`?